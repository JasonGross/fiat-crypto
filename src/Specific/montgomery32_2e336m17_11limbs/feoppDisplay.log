opp = λ var : Z.Syntax.base_type → Type,
λ '(x18, x19, x17, x15, x13, x11, x9, x7, x5, x3, x1)%core,
uint32_t x21, uint8_t x22 = subborrow_u32(0x0, 0x0, x1);
uint32_t x24, uint8_t x25 = subborrow_u32(x22, 0x0, x3);
uint32_t x27, uint8_t x28 = subborrow_u32(x25, 0x0, x5);
uint32_t x30, uint8_t x31 = subborrow_u32(x28, 0x0, x7);
uint32_t x33, uint8_t x34 = subborrow_u32(x31, 0x0, x9);
uint32_t x36, uint8_t x37 = subborrow_u32(x34, 0x0, x11);
uint32_t x39, uint8_t x40 = subborrow_u32(x37, 0x0, x13);
uint32_t x42, uint8_t x43 = subborrow_u32(x40, 0x0, x15);
uint32_t x45, uint8_t x46 = subborrow_u32(x43, 0x0, x17);
uint32_t x48, uint8_t x49 = subborrow_u32(x46, 0x0, x19);
uint32_t x51, uint8_t x52 = subborrow_u32(x49, 0x0, x18);
uint32_t x53 = cmovznz32(x52, 0x0, 0xffffffff);
uint32_t x55, uint8_t x56 = addcarryx_u32(0x0, x21, (x53 & 0xffffffef));
uint32_t x58, uint8_t x59 = addcarryx_u32(x56, x24, (x53 & 0xffffffff));
uint32_t x61, uint8_t x62 = addcarryx_u32(x59, x27, (x53 & 0xffffffff));
uint32_t x64, uint8_t x65 = addcarryx_u32(x62, x30, (x53 & 0xffffffff));
uint32_t x67, uint8_t x68 = addcarryx_u32(x65, x33, (x53 & 0xffffffff));
uint32_t x70, uint8_t x71 = addcarryx_u32(x68, x36, (x53 & 0xffffffff));
uint32_t x73, uint8_t x74 = addcarryx_u32(x71, x39, (x53 & 0xffffffff));
uint32_t x76, uint8_t x77 = addcarryx_u32(x74, x42, (x53 & 0xffffffff));
uint32_t x79, uint8_t x80 = addcarryx_u32(x77, x45, (x53 & 0xffffffff));
uint32_t x82, uint8_t x83 = addcarryx_u32(x80, x48, (x53 & 0xffffffff));
uint32_t x85, uint8_t _ = addcarryx_u32(x83, x51, (x53 & 0xffff));
(Return x85, Return x82, Return x79, Return x76, Return x73, Return x70, Return x67, Return x64, Return x61, Return x58, Return x55)
     : ∀ var : Z.Syntax.base_type → Type, expr Z.Syntax.base_type Z.Syntax.op (uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t -> uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t)

Argument scope is [function_scope]
