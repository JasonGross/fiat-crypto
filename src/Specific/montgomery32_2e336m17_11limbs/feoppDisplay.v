Require Import Crypto.Specific.montgomery32_2e336m17_11limbs.feopp.
Require Import Crypto.Specific.Framework.IntegrationTestDisplayCommon.

Print feopp.opp.
