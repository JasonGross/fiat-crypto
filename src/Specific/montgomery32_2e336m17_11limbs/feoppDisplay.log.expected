λ x : word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32,
Interp-η
(λ var : Syntax.base_type → Type,
 λ '(x19, x20, x18, x16, x14, x12, x10, x8, x6, x4, x2)%core,
 uint32_t x22, uint8_t/*bool*/ x23 = subborrow_u32(0x0, 0x0, x2);
 uint32_t x25, uint8_t/*bool*/ x26 = subborrow_u32(x23, 0x0, x4);
 uint32_t x28, uint8_t/*bool*/ x29 = subborrow_u32(x26, 0x0, x6);
 uint32_t x31, uint8_t/*bool*/ x32 = subborrow_u32(x29, 0x0, x8);
 uint32_t x34, uint8_t/*bool*/ x35 = subborrow_u32(x32, 0x0, x10);
 uint32_t x37, uint8_t/*bool*/ x38 = subborrow_u32(x35, 0x0, x12);
 uint32_t x40, uint8_t/*bool*/ x41 = subborrow_u32(x38, 0x0, x14);
 uint32_t x43, uint8_t/*bool*/ x44 = subborrow_u32(x41, 0x0, x16);
 uint32_t x46, uint8_t/*bool*/ x47 = subborrow_u32(x44, 0x0, x18);
 uint32_t x49, uint8_t/*bool*/ x50 = subborrow_u32(x47, 0x0, x20);
 uint32_t x52, uint8_t/*bool*/ x53 = subborrow_u32(x50, 0x0, x19);
 uint32_t x54 = cmovznz32(x53, 0x0, 0xffffffff);
 uint32_t x55 = (x54 & 0xffffffef);
 uint32_t x57, uint8_t/*bool*/ x58 = addcarryx_u32(0x0, x22, x55);
 uint32_t x59 = (x54 & 0xffffffff);
 uint32_t x61, uint8_t/*bool*/ x62 = addcarryx_u32(x58, x25, x59);
 uint32_t x63 = (x54 & 0xffffffff);
 uint32_t x65, uint8_t/*bool*/ x66 = addcarryx_u32(x62, x28, x63);
 uint32_t x67 = (x54 & 0xffffffff);
 uint32_t x69, uint8_t/*bool*/ x70 = addcarryx_u32(x66, x31, x67);
 uint32_t x71 = (x54 & 0xffffffff);
 uint32_t x73, uint8_t/*bool*/ x74 = addcarryx_u32(x70, x34, x71);
 uint32_t x75 = (x54 & 0xffffffff);
 uint32_t x77, uint8_t/*bool*/ x78 = addcarryx_u32(x74, x37, x75);
 uint32_t x79 = (x54 & 0xffffffff);
 uint32_t x81, uint8_t/*bool*/ x82 = addcarryx_u32(x78, x40, x79);
 uint32_t x83 = (x54 & 0xffffffff);
 uint32_t x85, uint8_t/*bool*/ x86 = addcarryx_u32(x82, x43, x83);
 uint32_t x87 = (x54 & 0xffffffff);
 uint32_t x89, uint8_t/*bool*/ x90 = addcarryx_u32(x86, x46, x87);
 uint32_t x91 = (x54 & 0xffffffff);
 uint32_t x93, uint8_t/*bool*/ x94 = addcarryx_u32(x90, x49, x91);
 uint32_t x95 = (x54 & 0xffff);
 uint32_t x97, uint8_t/*bool*/ _ = addcarryx_u32(x94, x52, x95);
 (Return x97, Return x93, Return x89, Return x85, Return x81, Return x77, Return x73, Return x69, Return x65, Return x61, Return x57))
x
     : word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 → ReturnType (uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t)
